lib/internal/bootstrap_node.js

```diff
@@ -268,15 +268,24 @@
     var console;
     Object.defineProperty(global, 'console', {
       configurable: true,
-      enumerable: true,
+      enumerable: false,
       get: function() {
         if (!console) {
           console = NativeModule.require('console');
           installInspectorConsoleIfNeeded(console,
                                           inspectorConsole,
                                           wrapConsoleCall);
         }
+        Object.defineProperty(global, 'console', {
+          configurable: true,
+          writable: true,
+          enumerable: false,
+          value: console
+        });
         return console;
+      },
+      set: function(customConsole) {
+        console = customConsole;
```

Comment:
What I meant was:

```js
// Initially, the `console` variable is `undefined`, since console will be lazily
// loaded in the getter.

global.console = undefined;
// global.console's setter is called; the `console` variable remains `undefined`.

assert.strictEqual(global.console, undefined);
// global.console's getter is called
// Since the `console` cache variable is `undefined` and therefore false-y,
// the getter still calls NativeModule.require() and returns the object
// obtained from it, instead of returning `undefined` as expected.
```